### PR TITLE
DEVX-1888: new confluent local syntax

### PR DESCRIPTION
**Pull request by brianstrauch**
Replaced all old `confluent local` commands across the entire repo.
Manually verified that the following demos work correctly by running their `start.sh` and `stop.sh` scripts:
- connect-streams-pipeline
- cp-quickstart
- microservices-orders
- music

Wait for https://github.com/confluentinc/cli/pull/547 before merging

### CODE REVIEW

File: utils/helper.sh
Comment (ybyzek) [inline utils/helper.sh:495]: What's the motivation of removing `check_confluent_local_service_ready` and adding `append_once`?

**Review by ybyzek — CHANGES_REQUESTED** [review]
@brianstrauch thank you *so* much for this PR, you rock!

I'm marking this PR as "Request changes" to intentionally block this until CLI/CP packaging is confirmed by @sdanduConf 

cc: @rspurgeon for FYI

File: utils/helper.sh
Comment (brianstrauch) [reply utils/helper.sh:495]: `check_confluent_local_service_ready` [was not being used anywhere](https://github.com/confluentinc/examples/search?q=check_confluent_local_service_ready&type=Code)
`append_once` prevents this from happening (after running examples more than once):

```
$ tail ksql-server.properties

# Uncomment and complete the following to enable KSQL's integration to the Confluent Schema Registry:
# ksql.schema.registry.url=http://localhost:8081
auto.offset.reset=earliest
auto.offset.reset=earliest
auto.offset.reset=earliest
auto.offset.reset=earliest
auto.offset.reset=earliest
auto.offset.reset=earliest
auto.offset.reset=earliest
```

Comment (ybyzek) [reply utils/helper.sh:495]: @brianstrauch great, thank you.

wrt `auto.offset.reset=earliest`:   @rspurgeon was also thinking to fix that as well (and might have had a PR open for that too?), cc-ing him here for additional review on that.

File: microservices-orders/start.sh
Comment (ybyzek) [inline microservices-orders/start.sh]: My guess is this line was commented out for quick validation, but please uncomment before PR merge.

File: microservices-orders/start.sh
Comment (ybyzek) [inline microservices-orders/start.sh]: My guess is this line was commented out for quick validation, but please uncomment before PR merge.

File: clients/cloud/confluent-cli/confluent-cli-ccsr-example.sh
Comment (ybyzek) [inline clients/cloud/confluent-cli/confluent-cli-ccsr-example.sh]: @brianstrauch Just curious if you validated this specific example?

File: microservices-orders/start.sh
Comment (brianstrauch) [reply microservices-orders/start.sh]: Yep, thanks for the catch.

File: clients/cloud/confluent-cli/confluent-cli-ccsr-example.sh
Comment (brianstrauch) [reply clients/cloud/confluent-cli/confluent-cli-ccsr-example.sh]: No, that should be `confluent local services kafka produce $topic_name`. I'll fix and validate.

File: utils/helper.sh
Comment (rspurgeon) [reply utils/helper.sh:495]: This looks good to me.  I had a similiar fix but it didn't make it in so let's definitely add this one.  Thanks @brianstrauch

**Review by sdandu-gh — APPROVED** [review]
LGTM. @ybyzek i went through all the demos in this PR end-to-end (with a couple changes that will follow-on) and confirmed the CLI pieces look good here.

**Review by ybyzek — CHANGES_REQUESTED** [review]
@brianstrauch the merge conflict arises due to a commit yesterday that moved the markdown file to an rst file in a different location.  So in addition to resolving the merge conflict (it should be https://github.com/confluentinc/examples/blob/6.0.x/clients/cloud/confluent-cli/README.md), the following file needs to be updated as well: https://github.com/confluentinc/examples/blob/6.0.x/clients/docs/confluent-cli.rst. This may necessitate the `brianstrauch:confluent-local-syntax` branch to be rebased off the latest `6.0.x` branch upstream.

File: connect-streams-pipeline/docs/index.rst
Comment (ybyzek) [inline connect-streams-pipeline/docs/index.rst:132]: Good catch @brianstrauch .  In this file there are two more references to `confluent local produce`, could you please update those as well?

Comment (brianstrauch) [reply connect-streams-pipeline/docs/index.rst:132]: Sure! I didn't change them originally because they looked like version notes, and I didn't know if they should be left unedited for historical purposes.

**Review by ybyzek — APPROVED** [review]
@brianstrauch thanks for proactive the PR, and @sdanduConf thanks for validating.  

LGTM.

**Pull request merged**
